internal/document/attributes.go

```diff
@@ -0,0 +1,219 @@
+package document
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+
+	"go.uber.org/multierr"
+	"golang.org/x/exp/slices"
+
+	"github.com/pelletier/go-toml"
+)
+
+type Attributes map[string]string
+
+type attributeParser interface {
+	ParseAttributes(raw []byte) (Attributes, error)
```

Comment:
As the name already includes `attribute`, it can be simplified:

```go
type attributeParser interface {
	Parse(raw []byte) (Attributes, error)
    Write(attr Attributes, w io.Writer) error
}
```